x-pack/plugins/uptime/server/lib/requests/get_monitor_list_pagination.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FetchMonitorListPaginationParams } from '../../../common/runtime_types';
+import { createEsQuery, UptimeESClient } from '../lib';
+import { QueryContext } from './search';
+import { CONTEXT_DEFAULTS } from '../../../common/constants';
+
+export type MonitorListPaginationResult = ReturnType<typeof getMonitorListPagination>;
+
+export const getMonitorListPagination = async ({
+  query,
+  filters,
+  statusFilter,
+  uptimeEsClient,
+  beforeMonitorId,
+  afterMonitorId,
+  dateRangeStart,
+  dateRangeEnd,
+}: FetchMonitorListPaginationParams & { uptimeEsClient: UptimeESClient }) => {
+  const boolFilters = filters ? JSON.parse(filters) : null;
+  const additionalFilters = [];
+
+  if (boolFilters) {
+    additionalFilters.push(boolFilters);
+  }
+  const parsedFilters = filters && filters !== '' ? JSON.parse(filters) : null;
+
+  const queryContext = new QueryContext(
+    uptimeEsClient,
+    dateRangeStart,
+    dateRangeEnd,
+    CONTEXT_DEFAULTS.CURSOR_PAGINATION,
+    parsedFilters,
+    10,
+    statusFilter,
+    query
+  );
+
+  const queryFilters = await queryContext.dateAndCustomFilters();
+
+  if (statusFilter) {
+    queryFilters.push({ match: { 'monitor.status': statusFilter } });
+  }
+
+  const params = createEsQuery({
+    body: {
+      query: {
+        bool: {
+          filter: [
+            ...queryFilters,
+            {
+              exists: {
+                field: 'summary',
+              },
+            },
+          ],
+          ...(query
+            ? {
+                minimum_should_match: 1,
+                should: [
+                  {
+                    multi_match: {
+                      query: escape(query),
+                      type: 'phrase_prefix' as const,
+                      fields: ['monitor.id.text', 'monitor.name.text', 'url.full.text'],
+                    },
+                  },
+                ],
+              }
+            : {}),
+        },
+      },
+      size: 0,
+      aggs: {
+        after: {
+          composite: {
+            size: 1,
+            after: {
+              monitorId: afterMonitorId,
+            },
+            sources: [
+              {
+                monitorId: {
+                  terms: {
+                    field: 'monitor.id',
+                  },
+                },
+              },
+            ],
+          },
+        },
+        before: {
```

Comment:
I was initially confused as to why we were returning the before and after aggregations this way. After investigating it, I can see that we are essentially using at as a boolean to check against the length and determine if there is a before and after, but not using the data in any way. Perhaps we can do some formatting on the data before we return it, to derive a boolean from the length and return it as `hasPrevious` and `hasNext`. Of course, it would make the api a little less straightforward as we'd have to format the data before returning it, but may make the component layer more straightforward.

Reply:
actually i do want to use this data in future to make number pagination, dependent upon design consideration